clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1121,8 +1130,12 @@ public Generation(int generationId, String memberId, String protocol) {
             this.protocol = protocol;
         }
 
-        public boolean isValid() {
-            return generationId != OffsetCommitRequest.DEFAULT_GENERATION_ID;
+        /**
+         * @return true if this generation has a valid member id. A generation might not be {@link #NO_GENERATION}, but
```

Comment:
This docstring could be improved. Perhaps we can just mention that we will may have a memberId before we are part of a group generation.